Method "LiveStripeResponseGetter.handleApiError()" throws unhandled exception "com.google.gson.JsonSyntaxException"

Java 8.0
stripe-java library 5.31.0 but the method implementation is currently the very same.

```
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
    at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:224)
    at com.google.gson.Gson.fromJson(Gson.java:887)
    at com.google.gson.Gson.fromJson(Gson.java:852)
    at com.google.gson.Gson.fromJson(Gson.java:801)
    at com.google.gson.Gson.fromJson(Gson.java:773)
    at com.stripe.net.LiveStripeResponseGetter.handleAPIError(LiveStripeResponseGetter.java:663)
    at com.stripe.net.LiveStripeResponseGetter._request(LiveStripeResponseGetter.java:508)
    at com.stripe.net.LiveStripeResponseGetter.request(LiveStripeResponseGetter.java:72)
    at com.stripe.net.APIResource.request(APIResource.java:169)
    at com.stripe.model.Plan.retrieve(Plan.java:219)
    at com.stripe.model.Plan.retrieve(Plan.java:181)
```

My suggestion is that this method should catch any thrown "com.google.gson.JsonSyntaxException" and handle it properly as expected.

@remi-stripe  I am using stripe-java version 3.2.0 library. We usually do not get this error, it looks like that it happens only on a specific time(i.e on 2018-09-12 00:00:XX) multiple times.
Logs:

Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:176)
	at com.google.gson.Gson.fromJson(Gson.java:803)
	at com.google.gson.Gson.fromJson(Gson.java:768)
	at com.google.gson.Gson.fromJson(Gson.java:717)
	at com.google.gson.Gson.fromJson(Gson.java:689)
	at com.stripe.net.LiveStripeResponseGetter.handleAPIError(LiveStripeResponseGetter.java:588)
	at com.stripe.net.LiveStripeResponseGetter._request(LiveStripeResponseGetter.java:444)
	at com.stripe.net.LiveStripeResponseGetter.request(LiveStripeResponseGetter.java:66)
	at com.stripe.net.APIResource.request(APIResource.java:144)
	at com.stripe.model.Customer.retrieve(Customer.java:371)
	at com.stripe.model.Customer.retrieve(Customer.java:257)

--- a/src/main/java/com/stripe/net/LiveStripeResponseGetter.java
+++ b/src/main/java/com/stripe/net/LiveStripeResponseGetter.java
@@ -1,5 +1,7 @@
 package com.stripe.net;
 
+import com.google.gson.JsonSyntaxException;
+
 import com.stripe.Stripe;
 import com.stripe.exception.ApiConnectionException;
 import com.stripe.exception.ApiException;
@@ -531,7 +533,12 @@ public class LiveStripeResponseGetter implements StripeResponseGetter {
       handleApiError(responseBody, responseCode, requestId);
     }
 
-    T resource = ApiResource.GSON.fromJson(responseBody, clazz);
+    T resource = null;
+    try {
+      resource = ApiResource.GSON.fromJson(responseBody, clazz);
+    } catch (JsonSyntaxException e) {
+      raiseMalformedJsonError(responseBody, responseCode, requestId);
+    }
 
     if (resource instanceof StripeObject) {
       StripeObject obj = (StripeObject)resource;
@@ -555,7 +562,12 @@ public class LiveStripeResponseGetter implements StripeResponseGetter {
       handleOAuthError(responseBody, responseCode, requestId);
     }
 
-    T resource = ApiResource.GSON.fromJson(responseBody, clazz);
+    T resource = null;
+    try {
+      resource = ApiResource.GSON.fromJson(responseBody, clazz);
+    } catch (JsonSyntaxException e) {
+      raiseMalformedJsonError(responseBody, responseCode, requestId);
+    }
 
     return resource;
   }
@@ -690,11 +702,25 @@ public class LiveStripeResponseGetter implements StripeResponseGetter {
 
   }
 
+  private static void raiseMalformedJsonError(String responseBody, int responseCode,
+      String requestId) throws ApiException {
+    throw new ApiException(
+      String.format(
+          "Invalid response object from API: %s. (HTTP response code was %d)",
+          responseBody, responseCode),
+      requestId, null, responseCode, null);
+  }
+
   private static void handleApiError(String responseBody, int responseCode, String requestId)
       throws ApiException, AuthenticationException, CardException, IdempotencyException,
       InvalidRequestException {
-    LiveStripeResponseGetter.StripeError error = ApiResource.GSON.fromJson(responseBody,
-        LiveStripeResponseGetter.StripeErrorContainer.class).error;
+    LiveStripeResponseGetter.StripeError error = null;
+    try {
+      error = ApiResource.GSON.fromJson(responseBody,
+          LiveStripeResponseGetter.StripeErrorContainer.class).error;
+    } catch (JsonSyntaxException e) {
+      raiseMalformedJsonError(responseBody, responseCode, requestId);
+    }
     switch (responseCode) {
       case 400:
       case 404:
@@ -723,8 +749,13 @@ public class LiveStripeResponseGetter implements StripeResponseGetter {
       throws InvalidClientException, InvalidGrantException,
       com.stripe.exception.oauth.InvalidRequestException, InvalidScopeException,
       UnsupportedGrantTypeException, UnsupportedResponseTypeException, ApiException {
-    LiveStripeResponseGetter.StripeOAuthError error = ApiResource.GSON.fromJson(responseBody,
-        LiveStripeResponseGetter.StripeOAuthError.class);
+    LiveStripeResponseGetter.StripeOAuthError error = null;
+    try {
+      error = ApiResource.GSON.fromJson(responseBody,
+          LiveStripeResponseGetter.StripeOAuthError.class);
+    } catch (JsonSyntaxException e) {
+      raiseMalformedJsonError(responseBody, responseCode, requestId);
+    }
     String code = error.error;
     String description = (error.errorDescription != null) ? error.errorDescription : code;
 

